src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -225,20 +225,16 @@ public CaseEvent createCaseEvent(
 
     switch (caseEvent.getCategory()) {
       case RESPONDENT_ENROLED:
-        List<CaseGroup> caseGroups =
-            caseGroupService.findCaseGroupsForExecutedCollectionExercises(targetCase);
-        processCaseCreationAndTransitionsDuringEnrolment(category, caseGroups, caseEvent, newCase);
-        break;
-      case SUCCESSFUL_RESPONSE_UPLOAD:
-      case COMPLETED_BY_PHONE:
-        createNewCase(category, caseEvent, targetCase, newCase);
-        updateAllAssociatedBiCases(targetCase, category);
+        processActionPlanChange(targetCase, true);
         break;
       case GENERATE_ENROLMENT_CODE:
-      case NO_ACTIVE_ENROLMENTS:
         replaceIAC(targetCase);
         effectTargetCaseStateTransition(category, targetCase);
         break;
+      case NO_ACTIVE_ENROLMENTS:
+        replaceIAC(targetCase);
+        processActionPlanChange(targetCase, false);
+        break;
       default:
         createNewCase(category, caseEvent, targetCase, newCase);
         effectTargetCaseStateTransition(category, targetCase);
```

Comment:
Rather than having this in the default we could have every case event goes through `effectTargetCaseStateTransition`
Similarly to `transitionCaseGroupStatus` it will only actually do anything for the relevant case events. This should reduce what we have to include in the switch block